sktime/forecasting/base/_adapters.py

```diff
@@ -2,20 +2,254 @@
+
+        # Add regressor (multivariate)
+        if X is not None:
+            df = df.merge(X, left_index=True, right_on=X.index)
+            for col in X.columns:
+                self._forecaster.add_regressor(col)
+
+        self._forecaster.fit(df=df, **fit_params)
+        self._is_fitted = True
+        return self
+
+    def update(self, y, X=None, update_params=False):
+        """Update fitted parameters
+
+        Parameters
+        ----------
+        y : pd.Series
+        X : pd.DataFrame
+        update_params : bool, optional (default=False)
+
+        Returns
+        -------
+        self : an instance of self
+        """
+        raise NotImplementedError()
+
+    def _predict(self, fh, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
+        """Predict
+
+        Parameters
+        ----------
+        fh : array-like
+            The forecasters horizon with the steps ahead to to predict.
+            Default is
+            one-step ahead forecast, i.e. np.array([1]).
+        X : pd.DataFrame, optional
+            Exogenous data, by default None
+        return_pred_int : bool, optional
+            Returns a pd.DataFrame with confidence intervalls, by default False
+        alpha : float, optional
+            Alpha level for confidence intervalls, by default DEFAULT_ALPHA
+
+        Returns
+        -------
+        y_pred : pandas.Series
+            Returns series of predicted values.
+
+        Raises
+        ------
+        Exception
+            Error when merging data
+        """
+        # neuralprophet in current version does not provide confidence interavals,
+        # will be added and then also included here.
+        if self.__class__.__name__ == "NeuralProphet" and return_pred_int:
+            raise NotImplementedError("Confidence interavals coming soon")
+
+        if alpha != DEFAULT_ALPHA:
+            raise NotImplementedError(
+                "alpha must be given in Prophet() as interval_width (1-alpha)"
+            )
+        fh = fh.to_relative(cutoff=self.cutoff)
+        if isinstance(fh.to_pandas(), pd.DatetimeIndex):
+            df = pd.DataFrame()
+            df["ds"] = fh.to_pandas()
+        else:
+            # Try to create pd.DatetimeIndex
+            df = self._coerce_to_datetime_index(fh=fh)
+
+        # Merge X with df (of created future DatetimeIndex values)
+        df = _merge_X(fh=fh, X=X, df=df)
+        out = self._forecaster.predict(df)
+
+        y_hat = "yhat"
+        if self.__class__.__name__ == "NeuralProphet":
+            y_hat = "yhat1"
+        y_in_sample = out[out["ds"] <= self.cutoff][y_hat]
+        y_out_sample = out[out["ds"] > self.cutoff][y_hat]
+        y_pred = self._get_y_pred(y_in_sample=y_in_sample, y_out_sample=y_out_sample)
+
+        if return_pred_int:
+            pred_int = self._get_pred_int(
+                lower=out["yhat_lower"], upper=out["yhat_upper"]
```

Comment:
this wouldnt work for neural prophet, the currently deployed version emits the following columns when calling forecast: `'ds', 'y', 'yhat1', 'residual1', 'trend', 'season_yearly', 'season_weekly'`.

Reply:
I see this is managed above with a `NotImplemented` so... ignore this